cli/server.go

```diff
@@ -393,7 +393,7 @@ func Server(vip *viper.Viper, newAPI func(context.Context, *coderd.Options) (*co
 					return xerrors.Errorf("OIDC issuer URL must be set!")
 				}
 
-				ctx, err := handleOauth2ClientCertificates(ctx, cfg)
+				ctx, err := codersdk.HandleOauth2ClientCertificates(ctx, cfg.TLS)
```

Comment:
This is an interesting issue. We're storing the client in the context, but we're also shadowing the parent context so this value will not get passed along to the server, only to `oidc.NewProvider`.

But even if we did not shadow `ctx` here, `ctx` is not used as a base-context by the HTTP server (AFAICT), so it would require some other changes to fix the issue this way.

Reply:
I suppose at the moment `oidc.NewProvider` is where my issue is

But I'm content for _all_ the calls coder makes to use these client certs (would probably prevent me from running into issues later since my organization protects _everything_ with client cert auth)

Like in my other comment, where is the best place for me to initialize this "base" http client once? Somewhere in `coderd.go`?

Maybe I should have asked before I started down this rabbit hole, but does Go have client cert flags like Java does?
```
-Djavax.net.ssl.keyStore=/path/to/serverkeystore.p12
-Djavax.net.ssl.keyStorePassword=password
```

That tells Java to use a _specific_ keystore (client certs) when making HTTP calls

Thanks!